(22,1,'2012-06-12 05:05:16','2012-06-12 05:05:16','I was just having some problems installing a FileZilla FTP server behind a firewalled NAT, and after considerable trial and error, the solution seems to be to make sure that the server is using a specific set of ports (e.g. 35000 - 35050) and forward those along with the traditional ports (20-21).','FileZilla server problems with a home NAT','','publish','open','open','','filezilla-server-problems-with-a-home-nat','','','2013-07-02 06:02:44','2013-07-02 06:02:44','',0,'http://andrewychoi.wordpress.com/?p=22',0,'post','',0),
(32,1,'2012-06-14 21:10:28','2012-06-14 21:10:28','I just spent an excruciating amount of time trying to figure out what was wrong with my install of Eclipse. It turns out that if you attempt to install Eclipse using just <code>sudo apt-get install eclipse</code> or <code>sudo apt-get install eclipse-pde</code>, you are unable to get Eclipse to add plugins. I ended up using the Software Center to install the Eclipse package with a Java development platform, and was then able to add the PyDev and Subclipse plugins to work!','Eclipse, PyDev, and Ubuntu','','publish','open','open','','eclipse-pydev-and-ubuntu','','','2013-07-03 03:24:04','2013-07-03 03:24:04','',0,'http://andrewychoi.wordpress.com/?p=32',0,'post','',0),
(44,1,'2012-06-19 19:21:29','2012-06-19 19:21:29','A few lessons learned today from working with MySQL variables, <code>WHILE</code> loops, and prepared statements:\r\n<ul>\r\n	<li>In a continuation statement for the <code>WHILE</code> loop, make sure to use user-defined variables (i.e. <code>DECLARE foo INT DEFAULT 10</code> as opposed to <code>SET @foo = 10</code>, which defines a session variable called <code>@foo</code> to be 10)</li>\r\n	<li>When using these user-defined variables, be sure to <code>DECLARE</code> them before any other code gets executed. I\'m embarrassed to say that this is one of the things that confounded me the most, despite having written more lines in C than in probably any other language.</li>\r\n</ul>\r\nMore to come as this project continues!','MySQL WHILE Loops and Different Types of Variables','','publish','open','open','','mysql-while-loops-and-different-types-of-variables','','','2013-07-03 03:22:22','2013-07-03 03:22:22','',0,'http://andrewychoi.wordpress.com/2012/06/19/mysql-while-loops-and-different-types-of-variables/',0,'post','',0),
(47,1,'2012-07-03 19:35:17','2012-07-03 19:35:17','Running across this bug is a bit odd, because I wasn\'t aware of any assertions that I was making, but this was the offending code:\n\n<code>{% if display_type == \'ts\' %}\n&lt;script src=\"/static/highstock.js\" type=\"text/javascript\"&gt;\n&lt;/script&gt;\n{% else if display_type == \'avg\' %}\n&lt;script src=\"/static/highcharts.js\" type=\"text/javascript\"&gt;\n&lt;/script&gt;\n{% endif %}</code>\n\nThe above code ended up causing an AssertionError: No exception supplied.\n\nThe idea was to do a conditional JavaScript import, but the problem actually ended up being in the Django template syntax.  The correction was to replace the \"else if\" with an \"elif\".  Rookie mistake!\n\nCorrected code:\n\n<code>{% if display_type == \'ts\' %}\n&lt;script src=\"/static/highstock.js\" type=\"text/javascript\"&gt;\n&lt;/script&gt;\n{% elif display_type == \'avg\' %}\n&lt;script src=\"/static/highcharts.js\" type=\"text/javascript\"&gt;\n&lt;/script&gt;\n{% endif %}</code>','AssertionError in Django templating','','publish','open','open','','assertionerror-no-exception-supplied-in-django-templating','','','2013-07-02 06:03:37','2013-07-02 06:03:37','',0,'http://andrewychoi.wordpress.com/?p=47',0,'post','',2),
(53,1,'2012-08-08 02:08:57','2012-08-08 02:08:57','On a project that I\'m working on currently, I have the following model definiton:\n\n<code>class Embassy(models.Model):\n    name = models.CharField(max_length = 200)\n    facebook_id = models.IntegerField(blank=True, null=True)\n    twitter_id = models.IntegerField(blank=True, null=True)\n    youtube_id = models.IntegerField(blank=True, null=True)\n</code>\n\nI had initially run a <code>python manage.py syncdb</code> before changing the IntegerFields to allow null values.  I thought that running a <code>python manage.py sqlclear embassy</code> would solve the problem, but it didn\'t actually drop the table that it needed to.  I had to go to <code>python manage.py dbshell</code> and run <code>DROP TABLE embassy_embassy</code> to get the table manually dropped.\n\nHopefully the wonderful people at Django will fix this bug soon!','Django issues with model fields','','publish','open','open','','django-issues-with-model-fields','','','2012-08-08 02:08:57','2012-08-08 02:08:57','',0,'http://andrewychoi.wordpress.com/?p=53',0,'post','',0),
(64,1,'2012-08-10 20:22:55','2012-08-10 20:22:55','I was debuggigng this bit of code today, and ran into a silly error:\n<code>\n# child_template.html:\n{% extends \"base.html\" %}\n\n{% block tab_contents %}\n{{ block.super }}\n{% endblock %}\n</code>\n\n<code>\n# base.html\n...\n{% for i in 123|make_list %}\n               <div id=\"tabs-{{ i }}\">\n                   {% if tab_index == i %}\n                    <div id=\"control\">\n	                     {% csrf_token %}\n	                        <table>\n	                            {{ form.as_table }}\n	                        </table>\n	                    \n	                    \n	                </div>\n                   {% endif %}\n               </div>\n{% endfor %}\n...\n</code>\n\nWhen you pass variables to a template, the type of the variable still does matter. It turns out that the <code>for i in 123|make_list</code> makes i a string variable.  Thus, if you want to conditionally print something, you need to pass <code>\'tab_index\': \'1\'</code> instead of <code>\'tab_index\': 1</code>.  It\'s a subtle distinction, but the \'==\' won\'t work if the variables are of disparate types.','Django templating: variables and conditionals','','publish','open','open','','django-templating-variables-and-conditionals','','','2012-08-10 20:22:55','2012-08-10 20:22:55','',0,'http://andrewychoi.wordpress.com/?p=64',0,'post','',0),
(71,1,'2012-12-04 06:11:00','2012-12-04 06:11:00','I was getting an <code>IntegrityError</code> while working with the <code>User</code> model in <code>django.contrib.auth.models</code>, with a call for a check to see if the <code>User</code> model\'s username was unique. I solved this by using a call to <code>User.objects.create_user()</code> instead of instantiating a new <code>User()</code> and manually setting all necessary attributes.\n\nAs a side note, if you\'re working with the <code>django.contrib.auth.forms UserCreationForm</code>, be aware that there won\'t be a <code>password</code> field in the returned form. There are instead <code>password1</code> and <code>password2</code> fields.','IntegrityError with Users in Django','','publish','open','open','','integrityerror-with-users-in-django','','','2012-12-04 06:11:00','2012-12-04 06:11:00','',0,'http://andrewychoi.wordpress.com/?p=71',0,'post','',0),
(79,1,'2013-03-21 06:44:08','2013-03-21 06:44:08','I mentioned to <a href=\"http://www.linkedin.com/pub/warren-mcquinn/27/190/aa5/\">a friend</a> today that I\'ve been making an effort to TeX the majority of my schoolwork, and I thought I\'d share a couple of things that I memorized in the interest of saving time:\n<h2><code>setcounter{secnumdepth}{-1}</code></h2>\n\nI tend to use the <code>section{}</code> keyword (function? delimiter?) a lot, especially when enumerating problem numbers, like so:\n\n<code>section{Problem 3}\nsubsection{a.}</code>\n\nor\n\n<code>section{3.4.2}</code>\n\nUsing <code>setcounter{secnumdepth}{-1}</code> helps me keep extraneous numbering to a minimum.\n\n<h2><code>usepackage{fullpage}</code></h2>\n\nPretty self-explanatory, I use this to make sure I\'m not wasting too much empty space.\n\n<h2><code>usepackage{amsmath}</code> and <code>begin{align}, end{align}</code></h2>\nI love the <code>amsmath</code> package and especially the <code>align</code> environment; it lets me get my equations in neat and tidy order. If you don\'t want numbered equations, make sure to use <code>begin{align*}</code> and <code>end{align*}</code>.\n\nI am, however, looking for a way of making sure that all columns in a three or more column structure are left-aligned. If you know how to do that, please let me know!\n\n<h2><code>includegraphics[width=textwidth]{filename.jpg}</code></h2>\n\nThe <code>includegraphics[]{}</code> function comes in the <code>graphicx</code> package (N.B. the function name is <code>include<b>graphics</b></code> but the package name is <code><b>graphicx</b></code>). The <code>[width=textwidth]</code> section is a great little hack to keep your image aspect ratio constant while ensuring that all of the image is visible.\n\nAny other great hacks out there?','LaTeX tips and tricks','','publish','open','open','','latex-tips-and-tricks','','','2013-03-21 06:44:08','2013-03-21 06:44:08','',0,'http://andrewychoi.wordpress.com/?p=79',0,'post','',0),
(94,1,'2013-04-23 04:08:39','0000-00-00 00:00:00','In retrospect, it should have been obvious that there was something wrong.\n\nMy father was a man of average height and build, a couple of pounds shy of portly. Salt and pepper hair was always a little bit unkempt, as if the tortoise-shell comb that he always kept on the bathroom counter was not quite up to the job. Thick glasses rimmed his face, the distortion that they perpetrated on his eyes evident despite the small surface of the lenses. He had arms that were very solidly white-collar, speaking to the backyard that was left untended and wild, in stark contrast to the manicured retaining walls of the neighbors to our right and left. The fence, comprised of thin, monolithic square bars, was also similarly contrasted; the sides that faced outwards were coated with fresh paint, while the sides that I and my mother could see had the fungus-like rust splotches reaching up from the earth.\n\nI loved our backyard. Its first twenty feet in its run backwards from the house was half patio, half lawn. As the lawn ended, the hill began to rise up, first obstructed by a retaining wall that was meticulously constructed, waving as if the force of the earth on its back was undulating. Further up the hill there was a second, less carefully placed wall, as if the builder had run out of bricks midway and wasn\'t willing to go back to the home improvement store and put these dirty bricks in his newly purchased minivan.\n\nBeyond the half done wall the hill took a mad rush up to the top of the backyard. There, at the top, the same off-white fence kept the rolling hills behind the house separate from us. It was always unclear what we were being protected from. The older kids on the block told us that they had seen wolves, and you had to listen with rapt attention or else you risked losing face in the block-wide games that happened on the weekends. Parents always talked fears of wolves down, and instead placed images of docile cows, or possibly a singular coyote.\n\nMy brother and I spent ','Draft - Utah','','draft','open','open','','','','','2013-04-23 04:08:39','0000-00-00 00:00:00','',0,'http://andrewychoi.wordpress.com/?p=94',0,'post','',0),
(95,1,'2013-05-19 04:52:56','2013-05-19 04:52:56','I recently decided to do a bit of finals procrastination by working on a problem that\'s been a mild nuisance for the past few months. My Google Contacts have had a problem with delineating first and last names, but that\'s a story for another post. If you\'re interested in the repo as it develops, check out <code>https://github.com/andrewychoi/fix_contacts</code>\r\n\r\nWhat I had been having trouble with was the consistent need to enter a username and password for every time that I wanted to <code>git push</code> to the server. I went into the GitHub user menu and added an SSH key, but access to my accounts was still problematic.\r\n\r\nThe problem was that my <code>project_directory/.git/config</code> file still told me that <code>origin</code> was to be reached over HTTPS. Changing that to SSH solved the protocol access problem. For exactness, I changed the line\r\n\r\n<code>https://github.com/username/project.git</code>\r\n\r\nto\r\n\r\n<code>ssh://github.com/username/project.git</code>\r\n\r\nNow, however, I was greeted with a\r\n\r\n<code>Permission denied (publickey).\r\nfatal: The remote end hung up unexpectedly</code>\r\n\r\nAt this point, I started searching the web for the solution to this problem. It turns out that I needed to SSH in as the <code>git</code> user. The fix was to change\r\n\r\n<code>ssh://github.com/username/project.git</code>\r\n\r\nto\r\n\r\n<code>ssh://git@github.com/username/project.git</code>\r\n\r\nNow everything works like a charm!\r\n\r\n<code>\r\n&gt; git pull\r\nAlready up-to-date.\r\n&gt; git push\r\nEverything up-to-date.\r\n</code>','Ensuring GitHub access over SSH','','publish','open','open','','ensuring-github-access-over-ssh','','','2013-07-03 03:34:58','2013-07-03 03:34:58','',0,'http://andrewychoi.wordpress.com/?p=95',0,'post','',0),
(108,1,'2013-07-01 00:47:37','2013-07-01 00:47:37','I\'m learning the lesson that I need to write these posts right after I run the installations; I\'m having trouble reconstructing what I did that got my machine to the point it is now (I tried replicating some of these steps on a clean 12.04 install, and for whatever reason I still can\'t get <code>virtualenv</code> to work correctly).\r\n\r\nIn any case, I decided to take a more proactive approach in moving the wide array of Python libraries to Python 3, and start building my personal projects using Py3k. I\'ll be posting periodically about new libraries that I\'ve helped port over; I expect that the majority of them will be simple <code>2to3</code> conversions and then checking.\r\n\r\nIn no particular order: steps that I took to get a <code>virtualenv</code> with Python 3.3 working on Ubuntu 12.04:\r\n\r\n<ul>\r\n	<li>Install Felix Krull\'s PPA for Python versions: <code>sudo add-apt-repository ppa:fkrull/deadsnakes</code></li>\r\n	<li>Install Python 3.3: <code>sudo aptitude install python3.3</code></li>\r\n	<li>Install <code>pip</code> for Python 3.3: <code>sudo aptitude install python3-setuptools</code></li>\r\n	<li>Create the <code>virtualenv</code>: <code>virtualenv -p /usr/bin/python3.3 target_directory</code></li>\r\n</ul>\r\n\r\nIf I come up with something else that I did to solve this problem, I\'ll be sure to update!','Python 3.3 with virtualenv and Ubuntu 12.04','','publish','open','open','','python-3-3-with-virtualenv-and-ubuntu-12-04','','','2013-07-03 03:47:51','2013-07-03 03:47:51','',0,'http://andrewychoi.wordpress.com/?p=108',0,'post','',0),
(121,1,'2013-07-01 00:47:37','2013-07-01 00:47:37','<p>Steps to take to get a virtualenv with Python 3.3 working on Ubuntu 12.04:</p>','Python 3.3 with virtualenv and Ubuntu 12.04','','inherit','open','open','','108-revision','','','2013-07-01 00:47:37','2013-07-01 00:47:37','',108,'http://sleepy-oasis-8661.herokuapp.com/?p=121',0,'revision','',0),
(131,1,'2012-06-12 05:05:16','2012-06-12 05:05:16','I was just having some problems installing a FileZilla FTP server behind a firewalled NAT, and after considerable trial and error, the solution seems to be to make sure that the server is using a specific set of ports (e.g. 35000 - 35050) and forward those along with the traditional ports (20-21).','FileZilla server problems with a home NAT','','inherit','open','open','','22-revision','','','2012-06-12 05:05:16','2012-06-12 05:05:16','',22,'http://www.andrewychoi.com/?p=131',0,'revision','',0),
(141,1,'2012-06-19 19:21:29','2012-06-19 19:21:29','<p>A few lessons learned today from working with MySQL variables, WHILE loops, and prepared statements:</p><ul><li>In a continuation statement for the WHILE loop, make sure to use user-defined variables (i.e. \"DECLARE foo INT DEFAULT 10\" as opposed to \"SET @foo = 10\", which defines a session variable called @foo to be 10)</li><li>When using these user-defined variables, be sure to DECLARE them before any other code gets executed.  I\'m embarrassed to say that this is one of the things that confounded me the most, despite having written more lines in C than in probably any other language.</li></ul><p>More to come as this project continues!</p>','MySQL WHILE Loops and Different Types of Variables','','inherit','open','open','','44-revision','','','2012-06-19 19:21:29','2012-06-19 19:21:29','',44,'http://www.andrewychoi.com/?p=141',0,'revision','',0),
(151,1,'2012-06-14 21:10:28','2012-06-14 21:10:28','I just spent an excruciating amount of time trying to figure out what was wrong with my install of Eclipse.  It turns out that if you attempt to install Eclipse using just sudo apt-get install eclipse or sudo apt-get install eclipse-pde, you are unable to get Eclipse to add plugins.  I ended up using the Software Center to install the Eclipse package with a Java development platform, and was then able to add the PyDev and Subclipse plugins to work!','Eclipse, PyDev, and Ubuntu','','inherit','open','open','','32-revision','','','2012-06-14 21:10:28','2012-06-14 21:10:28','',32,'http://www.andrewychoi.com/?p=151',0,'revision','',0),
(161,1,'2012-07-03 19:35:17','2012-07-03 19:35:17','Running across this bug is a bit odd, because I wasn\'t aware of any assertions that I was making, but this was the offending code:\n\n<code>{% if display_type == \'ts\' %}\n&lt;script src=\"/static/highstock.js\" type=\"text/javascript\"&gt;\n&lt;/script&gt;\n{% else if display_type == \'avg\' %}\n&lt;script src=\"/static/highcharts.js\" type=\"text/javascript\"&gt;\n&lt;/script&gt;\n{% endif %}</code>\n\nThe above code ended up causing an AssertionError: No exception supplied.\n\nThe idea was to do a conditional JavaScript import, but the problem actually ended up being in the Django template syntax.  The correction was to replace the \"else if\" with an \"elif\".  Rookie mistake!\n\nCorrected code:\n\n<code>{% if display_type == \'ts\' %}\n&lt;script src=\"/static/highstock.js\" type=\"text/javascript\"&gt;\n&lt;/script&gt;\n{% elif display_type == \'avg\' %}\n&lt;script src=\"/static/highcharts.js\" type=\"text/javascript\"&gt;\n&lt;/script&gt;\n{% endif %}</code>','AssertionError in Django templating','','inherit','open','open','','47-revision','','','2012-07-03 19:35:17','2012-07-03 19:35:17','',47,'http://www.andrewychoi.com/?p=161',0,'revision','',0),
(171,1,'2013-07-02 05:34:48','2013-07-02 05:34:48','This is an example page. It\'s different from a blog post because it will stay in one place and will show up in your site navigation (in most themes). Most people start with an About page that introduces them to potential site visitors. It might say something like this:\n\n<blockquote>Hi there! I\'m a bike messenger by day, aspiring actor by night, and this is my blog. I live in Los Angeles, have a great dog named Jack, and I like pi&#241;a coladas. (And gettin\' caught in the rain.)</blockquote>\n\n...or something like this:\n\n<blockquote>The XYZ Doohickey Company was founded in 1971, and has been providing quality doohickeys to the public ever since. Located in Gotham City, XYZ employs over 2,000 people and does all kinds of awesome things for the Gotham community.</blockquote>\n\nAs a new WordPress user, you should go to <a href=\"https://sleepy-oasis-8661.herokuapp.com/wp-admin/\">your dashboard</a> to delete this page and create new pages for your content. Have fun!','Sample Page','','inherit','open','open','','11-revision','','','2013-07-02 05:34:48','2013-07-02 05:34:48','',11,'http://www.andrewychoi.com/?p=171',0,'revision','',0),
(181,1,'2013-07-02 06:03:01','2013-07-02 06:03:01','<p>A few lessons learned today from working with MySQL variables, WHILE loops, and prepared statements:</p><ul><li>In a continuation statement for the WHILE loop, make sure to use user-defined variables (i.e. \"DECLARE foo INT DEFAULT 10\" as opposed to \"SET @foo = 10\", which defines a session variable called @foo to be 10)</li><li>When using these user-defined variables, be sure to DECLARE them before any other code gets executed.  I\'m embarrassed to say that this is one of the things that confounded me the most, despite having written more lines in C than in probably any other language.</li></ul><p>More to come as this project continues!</p>','MySQL WHILE Loops and Different Types of Variables','','inherit','open','open','','44-revision-2','','','2013-07-02 06:03:01','2013-07-02 06:03:01','',44,'http://www.andrewychoi.com/2013/44-revision-2/',0,'revision','',0),
(191,1,'2013-07-03 03:22:07','2013-07-03 03:22:07','A few lessons learned today from working with MySQL variables, <code>WHILE</code> loops, and prepared statements:\r\n<ul>\r\n	<li>In a continuation statement for the <code>WHILE</code> loop, make sure to use user-defined variables (i.e. <code>DECLARE foo INT DEFAULT 10</code> as opposed to <code>SET @foo = 10</code>, which defines a session variable called @foo to be 10)</li>\r\n	<li>When using these user-defined variables, be sure to <code>DECLARE</code> them before any other code gets executed. I\'m embarrassed to say that this is one of the things that confounded me the most, despite having written more lines in C than in probably any other language.</li>\r\n</ul>\r\nMore to come as this project continues!','MySQL WHILE Loops and Different Types of Variables','','inherit','open','open','','44-revision-3','','','2013-07-03 03:22:07','2013-07-03 03:22:07','',44,'http://www.andrewychoi.com/2013/44-revision-3/',0,'revision','',0),
(201,1,'2013-07-03 03:22:32','2013-07-03 03:22:32','I just spent an excruciating amount of time trying to figure out what was wrong with my install of Eclipse. It turns out that if you attempt to install Eclipse using just sudo apt-get install eclipse or sudo apt-get install eclipse-pde, you are unable to get Eclipse to add plugins. I ended up using the Software Center to install the Eclipse package with a Java development platform, and was then able to add the PyDev and Subclipse plugins to work!','Eclipse, PyDev, and Ubuntu','','inherit','open','open','','32-autosave','','','2013-07-03 03:22:32','2013-07-03 03:22:32','',32,'http://www.andrewychoi.com/2013/32-autosave/',0,'revision','',0),
(211,1,'2013-07-03 03:22:34','2013-07-03 03:22:34','I was just having some problems installing a FileZilla FTP server behind a firewalled NAT, and after considerable trial and error, the solution seems to be to make sure that the server is using a specific set of ports (e.g. 35000 - 35050) and forward those along with the traditional ports (20-21).','FileZilla server problems with a home NAT','','inherit','open','open','','22-autosave','','','2013-07-03 03:22:34','2013-07-03 03:22:34','',22,'http://www.andrewychoi.com/2013/22-autosave/',0,'revision','',0),
(221,1,'2013-07-02 06:03:17','2013-07-02 06:03:17','I just spent an excruciating amount of time trying to figure out what was wrong with my install of Eclipse.  It turns out that if you attempt to install Eclipse using just sudo apt-get install eclipse or sudo apt-get install eclipse-pde, you are unable to get Eclipse to add plugins.  I ended up using the Software Center to install the Eclipse package with a Java development platform, and was then able to add the PyDev and Subclipse plugins to work!','Eclipse, PyDev, and Ubuntu','','inherit','open','open','','32-revision-2','','','2013-07-02 06:03:17','2013-07-02 06:03:17','',32,'http://www.andrewychoi.com/2013/32-revision-2/',0,'revision','',0),
(231,1,'2013-07-03 03:22:53','2013-07-03 03:22:53','I just spent an excruciating amount of time trying to figure out what was wrong with my install of Eclipse. It turns out that if you attempt to install Eclipse using just <code>sudo apt-get install eclipse</code> or <code>sudo apt-get install eclipse-pde</code>, you are unable to get Eclipse to add plugins. I ended up using the Software Center to install the Eclipse package with a Java development platform, and was then able to add the PyDev and Subclipse plugins to work!','Eclipse, PyDev, and Ubuntu','','inherit','open','open','','32-revision-3','','','2013-07-03 03:22:53','2013-07-03 03:22:53','',32,'http://www.andrewychoi.com/2013/32-revision-3/',0,'revision','',0),
(241,1,'2013-05-19 04:52:56','2013-05-19 04:52:56','I recently decided to do a bit of finals procrastination by working on a problem that\'s been a mild nuisance for the past few months. My Google Contacts have had a problem with delineating first and last names, but that\'s a story for another post. If you\'re interested in the repo as it develops, check out https://github.com/andrewychoi/fix_contacts\n\nWhat I had been having trouble with was the consistent need to enter a username and password for every time that I wanted to <code>git push</code> to the server. I went into the GitHub user menu and added an SSH key, but access to my accounts was still problematic.\n\nThe problem was that my <code>project_directory/.git/config</code> file still told me that <code>origin</code> was to be reached over HTTPS. Changing that to SSH solved the protocol access problem. For exactness, I changed the line\n\n<code>https://github.com/username/project.git</code>\n\nto\n\n<code>ssh://github.com/username/project.git</code>\n\nNow, however, I was greeted with a\n\n<code>Permission denied (publickey).\nfatal: The remote end hung up unexpectedly</code>\n\nAt this point, I started searching the web for the solution to this problem. It turns out that I needed to SSH in as the <code>git</code> user. The fix was to change\n\n<code>ssh://github.com/username/project.git</code>\n\nto\n\n<code>ssh://git@github.com/username/project.git</code>\n\nNow everything works like a charm!\n\n<code>\n&gt; git pull\nAlready up-to-date.\n&gt; git push\nEverything up-to-date.\n</code>','Ensuring GitHub access over SSH','','inherit','open','open','','95-revision','','','2013-05-19 04:52:56','2013-05-19 04:52:56','',95,'http://www.andrewychoi.com/2013/95-revision/',0,'revision','',0),
(251,1,'2013-07-03 03:47:32','2013-07-03 03:47:32','I\'m learning the lesson that I need to write these posts right after I run the installations; I\'m having trouble reconstructing what I did that got my machine to the point it is now (I tried replicating some of these steps on a clean 12.04 install, and for whatever reason I still can\'t get <code>virtualenv</code> to work correctly).\n\nIn any case, I decided to take a more proactive approach in moving the wide array of Python libraries to Python 3, and start building my personal projects using Py3k. I\'ll be posting periodically about new libraries that I\'ve helped port over; I expect that the majority of them will be \n\nIn no particular order: steps that I took to get a <code>virtualenv</code> with Python 3.3 working on Ubuntu 12.04:\n\n<ul>\n	<li>Install Felix Krull\'s PPA for Python versions: <code>sudo add-apt-repository ppa:fkrull/deadsnakes</code></li>\n	<li>Install Python 3.3: <code>sudo aptitude install python3.3</code></li>\n	<li>Install <code>pip</code> for Python 3.3: <code>sudo aptitude install python3-setuptools</code></li>\n	<li>Create the <code>virtualenv</code>: <code>virtualenv -p /usr/bin/python3.3 target_directory</code></li>\n</ul>\n\nIf I come up with something else that I did to solve this problem, I\'ll be sure to update!','Python 3.3 with virtualenv and Ubuntu 12.04','','inherit','open','open','','108-revision-2','','','2013-07-03 03:47:32','2013-07-03 03:47:32','',108,'http://www.andrewychoi.com/2013/108-revision-2/',0,'revision','',0),
(261,1,'2013-07-03 00:17:40','2013-07-03 04:17:40','I got the hankering to try converting this blog to a static site (a la Jekyll, which my Ruby friends keep bringing up) and did some searching for Python equivalents. Putting aside Hyde (which I actually found by just searching \"hyde static site generator\" on a whim, without realizing that the Hyde project actually existed) the best thing for me seemed to be Pelican, which has a great migration tool (<code>pelican-import</code>) for migrating from Wordpress.\r\n\r\nIn terms of getting set up, there were a bunch of dependencies that needed to be satisfied; here are just a few of the ones that seemed to trip me up (the rest of them are a simple <code>pip install beautifulsoup4</code> or similar).\r\n\r\n<ul>\r\n	<li><code>pandoc</code> is a bit tricky, you need to install it via <code>sudo aptitude install pandoc</code> as opposed to a <code>pip</code> package.</li>\r\n	<li>Trying to run the importer: <code>pelican-import --wpfiles wp.file.xml</code> failed with:\r\n\r\n<blockquote>\r\n...\r\nbs4.FeatureNotFound: Couldn\'t find a tree builder with the features you requested: xml. Do you need to install a parser library?</blockquote>\r\n\r\nwhich prompted an attempt to install <code>lxml</code>, via <code>pip install lxml</code>, which then failed with\r\n\r\n<blockquote>...\r\n/home/andrew/workspace/personal-site/build/lxml/src/lxml/includes/etree_defs.h:9:31: fatal error: libxml/xmlversion.h: No such file or directory</blockquote>\r\n\r\nwhich I did some Googling to find that the solution was to run <code>sudo aptitude install libxml2-dev libxslt-dev</code> after which rerunning <code>pip install lxml</code> was enough to get the importer working!</li>\r\n</ul>','Pelican on Ubuntu 12.04 and Python 3.3','','publish','open','open','','pelican-on-ubuntu-12-04-and-python-3-3','','','2013-07-03 00:17:40','2013-07-03 04:17:40','',0,'http://www.andrewychoi.com/?p=261',0,'post','',0),
(271,1,'2013-07-03 00:16:07','2013-07-03 04:16:07','I got the hankering to try converting this blog to a static site (a la Jekyll, which my Ruby friends keep bringing up) and did some searching for Python equivalents. Putting aside Hyde (which I actually found by just searching \"hyde static site generator\" on a whim, without realizing that the Hyde project actually existed) the best thing for me seemed to be Pelican, which has a great migration tool (<code>pelican-import</code>) for migrating from Wordpress.\n\nIn terms of getting set up, there were a bunch of dependencies that needed to be satisfied; here are just a few of the ones that seemed to trip me up (the rest of them are a simple <code>pip install beautifulsoup4</code> or similar).\n\n<ul>\n	<li><code>pandoc</code> is a bit tricky, you need to install it via <code>sudo aptitude install pandoc</code> as opposed to a <code>pip</code> package.</li>\n	<li>Trying to run the importer: <code>pelican-import --wpfiles wp.file.xml</code> failed with:\n\n<blockquote>\n...\nbs4.FeatureNotFound: Couldn\'t find a tree builder with the features you requested: xml. Do you need to install a parser library?</blockquote>\n\nwhich prompted an attempt to install <code>lxml</code>, via <code>pip install lxml</code>, which then failed with\n\n<blockquote>...\n/home/andrew/workspace/personal-site/build/lxml/src/lxml/includes/etree_defs.h:9:31: fatal error: libxml/xmlversion.h: No such file or directory</blockquote>\n\nwhich I did some Googling to find that \nsudo aptitude install libxml2-dev libxslt-dev\n\nthen rerunning solved it','Pelican on Ubuntu 12.04 and Python 3.3','','inherit','open','open','','261-revision-v1','','','2013-07-03 00:16:07','2013-07-03 04:16:07','',261,'http://www.andrewychoi.com/2013/261-revision/',0,'revision','',0),
(281,1,'2013-07-03 00:16:36','2013-07-03 04:16:36','I got the hankering to try converting this blog to a static site (a la Jekyll, which my Ruby friends keep bringing up) and did some searching for Python equivalents. Putting aside Hyde (which I actually found by just searching \"hyde static site generator\" on a whim, without realizing that the Hyde project actually existed) the best thing for me seemed to be Pelican, which has a great migration tool (<code>pelican-import</code>) for migrating from Wordpress.\r\n\r\nIn terms of getting set up, there were a bunch of dependencies that needed to be satisfied; here are just a few of the ones that seemed to trip me up (the rest of them are a simple <code>pip install beautifulsoup4</code> or similar).\r\n\r\n<ul>\r\n	<li><code>pandoc</code> is a bit tricky, you need to install it via <code>sudo aptitude install pandoc</code> as opposed to a <code>pip</code> package.</li>\r\n	<li>Trying to run the importer: <code>pelican-import --wpfiles wp.file.xml</code> failed with:\r\n\r\n<blockquote>\r\n...\r\nbs4.FeatureNotFound: Couldn\'t find a tree builder with the features you requested: xml. Do you need to install a parser library?</blockquote>\r\n\r\nwhich prompted an attempt to install <code>lxml</code>, via <code>pip install lxml</code>, which then failed with\r\n\r\n<blockquote>...\r\n/home/andrew/workspace/personal-site/build/lxml/src/lxml/includes/etree_defs.h:9:31: fatal error: libxml/xmlversion.h: No such file or directory</blockquote>\r\n\r\nwhich I did some Googling to find that the solution was to run <code>sudo aptitude install libxml2-dev libxslt-dev</code> after which rerunning <code>pip install lxml</code> was enough to get the importer working!','Pelican on Ubuntu 12.04 and Python 3.3','','inherit','open','open','','261-revision-v1','','','2013-07-03 00:16:36','2013-07-03 04:16:36','',261,'http://www.andrewychoi.com/2013/261-revision-2/',0,'revision','',0),
(291,1,'2013-07-03 00:17:01','2013-07-03 04:17:01','I got the hankering to try converting this blog to a static site (a la Jekyll, which my Ruby friends keep bringing up) and did some searching for Python equivalents. Putting aside Hyde (which I actually found by just searching \"hyde static site generator\" on a whim, without realizing that the Hyde project actually existed) the best thing for me seemed to be Pelican, which has a great migration tool (<code>pelican-import</code>) for migrating from Wordpress.\r\n\r\nIn terms of getting set up, there were a bunch of dependencies that needed to be satisfied; here are just a few of the ones that seemed to trip me up (the rest of them are a simple <code>pip install beautifulsoup4</code> or similar).\r\n\r\n<ul>\r\n	<li><code>pandoc</code> is a bit tricky, you need to install it via <code>sudo aptitude install pandoc</code> as opposed to a <code>pip</code> package.</li>\r\n	<li>Trying to run the importer: <code>pelican-import --wpfiles wp.file.xml</code> failed with:\r\n\r\n<blockquote>\r\n...\r\nbs4.FeatureNotFound: Couldn\'t find a tree builder with the features you requested: xml. Do you need to install a parser library?</blockquote>\r\n\r\nwhich prompted an attempt to install <code>lxml</code>, via <code>pip install lxml</code>, which then failed with\r\n\r\n<blockquote>...\r\n/home/andrew/workspace/personal-site/build/lxml/src/lxml/includes/etree_defs.h:9:31: fatal error: libxml/xmlversion.h: No such file or directory</blockquote>\r\n\r\nwhich I did some Googling to find that the solution was to run <code>sudo aptitude install libxml2-dev libxslt-dev</code> after which rerunning <code>pip install lxml</code> was enough to get the importer working!</li>\r\n</ul>','Pelican on Ubuntu 12.04 and Python 3.3','','inherit','open','open','','261-revision-v1','','','2013-07-03 00:17:01','2013-07-03 04:17:01','',261,'http://www.andrewychoi.com/2013/261-revision-3/',0,'revision','',0),
(301,1,'2013-07-03 00:26:46','2013-07-03 04:26:46','As I mentioned previously, I\'m trying to do as many of my projects with Python 3 as possible going forward, which is going to involve porting a bunch of libraries to work with Python 3.3 (I\'m using 3.3 because <code>simplejson</code> doesn\'t support 3.2, and I enjoy working with JSON much more than XML). Today\'s library is the <code>python-linkedin</code> API access library, <a href=\"http://developer.linkedin.com/documents/libraries-and-tools\">mentioned by LinkedIn themselves</a> under \"More OAuth Libraries\" and <a href=\"https://github.com/ozgur/python-linkedin\">hosted on GitHub</a>.\r\n\r\nMy port to Python 3 can be found at <a href=\"https://github.com/andrewychoi/python3-linkedin\">https://github.com/andrewychoi/python3-linkedin</a>. As issues arise, please let me know!','LinkedIn API library ported to Python 3.3','','publish','open','open','','linkedin-api-library-ported-to-python-3-3','','','2013-07-03 00:26:46','2013-07-03 04:26:46','',0,'http://www.andrewychoi.com/?p=301',0,'post','',4),
(311,1,'2013-07-03 00:25:31','2013-07-03 04:25:31','As I mentioned previously, I\'m trying to do as many of my projects with Python 3 as possible going forward, which is going to involve porting a bunch of libraries to work with Python 3.3 (I\'m using 3.3 because <code>simplejson</code> doesn\'t support 3.2, and I enjoy working with JSON much more than XML). Today\'s library is the <code>python-linkedin</code> API access library, <a href=\"http://developer.linkedin.com/documents/libraries-and-tools\">mentioned by LinkedIn themselves</a> under \"More OAuth Libraries\" and <a href=\"https://github.com/ozgur/python-linkedin\">hosted on GitHub</a>.\n\nMy port to Python 3 can be found at <a href=\"https://github.com/andrewychoi/python3-linkedin\">https://github.com/andrewychoi/python3-linkedin</a>. As issues arise, please let me know!','LinkedIn API library in Python 3.3','','inherit','open','open','','301-revision','','','2013-07-03 00:25:31','2013-07-03 04:25:31','',301,'http://www.andrewychoi.com/2013/301-revision/',0,'revision','',0),
(321,1,'2013-07-03 00:26:15','2013-07-03 04:26:15','As I mentioned previously, I\'m trying to do as many of my projects with Python 3 as possible going forward, which is going to involve porting a bunch of libraries to work with Python 3.3 (I\'m using 3.3 because <code>simplejson</code> doesn\'t support 3.2, and I enjoy working with JSON much more than XML). Today\'s library is the <code>python-linkedin</code> API access library, <a href=\"http://developer.linkedin.com/documents/libraries-and-tools\">mentioned by LinkedIn themselves</a> under \"More OAuth Libraries\" and <a href=\"https://github.com/ozgur/python-linkedin\">hosted on GitHub</a>.\r\n\r\nMy port to Python 3 can be found at <a href=\"https://github.com/andrewychoi/python3-linkedin\">https://github.com/andrewychoi/python3-linkedin</a>. As issues arise, please let me know!','LinkedIn API library in Python 3.3','','inherit','open','open','','301-revision-2','','','2013-07-03 00:26:15','2013-07-03 04:26:15','',301,'http://www.andrewychoi.com/2013/301-revision-2/',0,'revision','',0),
(341,1,'2013-07-03 16:49:50','0000-00-00 00:00:00','I mentioned in a previous post that I was having trouble getting a <code>virtualenv</code> to work with Python 3, and I think I\'ve found the issue. The machine that wasn\'t installing an up-to-date version of <code>setuptools</code> (and thus <code>pip</code>) was running <code>virtualenv 1.7</code>, where the current version is <code>1.9.1</code>. Somewhere along the upgrade chain, <code>virtualenv</code> now intelligently identifies the version of <code>pip</code> that you need, and if you don\'t have it, installs it. Now my <code>virtualenv</code> is working great, and I can get back to the business of working to write an IMAP inbox cleaner for an email tool that we use at work.','Virtualenv not installing pip for Python 3.3','','draft','open','open','','','','','2013-07-03 16:49:50','2013-07-03 20:49:50','',0,'http://www.andrewychoi.com/?p=341',0,'post','',0),
(351,1,'2013-07-03 16:48:56','2013-07-03 20:48:56','I mentioned in a previous post that I was having trouble getting a virtualenv to work with Python 3, and I think I\'ve found the issue. The machine that wasn\'t installing an up-to-date version of setuptools (and thus pip) was running virtualenv 1.7, where the current version is 1.9.1. Somewhere along the upgrade chain, virtualenv now intelligently identifies the version of pip that you need, and if you don\'t have it, installs it. Now my virtualenv is working great, and I can get back to the business of working ','Virtualenv not installing pip for Python 3.3','','inherit','open','open','','341-revision','','','2013-07-03 16:48:56','2013-07-03 20:48:56','',341,'http://www.andrewychoi.com/2013/341-revision/',0,'revision','',0),
(371,1,'2013-10-21 21:06:16','2013-10-22 01:06:16','Quick tip: if you\'re having problems with installing Ubuntu 13.04+ on a Dell XPS 14, make sure you\'re using the AMD 64 image (ubuntu-13.04-desktop-amd64.iso) and not the x86 image. That way, you get EFI boot capability.\r\n\r\n&nbsp;\r\n\r\nWill expand on this further at a later time.','Installing Ubuntu 13.04 on Dell XPS 14','','publish','open','open','','installing-ubuntu-13-04-on-dell-xps-14','','','2013-10-21 21:06:16','2013-10-22 01:06:16','',0,'http://www.andrewychoi.com/?p=371',0,'post','',0),
(381,1,'2013-10-21 21:05:35','2013-10-22 01:05:35','Quick tip: if you\'re having problems with installing Ubuntu 13.04+ on a Dell XPS 14, make sure you\'re using the AMD 64 image (','Installing Ubuntu 13.04 on Dell XPS 14','','inherit','open','open','','371-revision','','','2013-10-21 21:05:35','2013-10-22 01:05:35','',371,'http://www.andrewychoi.com/2013/371-revision/',0,'revision','',0),
(391,1,'2013-10-21 21:07:21','2013-10-22 01:07:21','Quick tip: if you\'re having problems with installing Ubuntu 13.04+ on a Dell XPS 14, make sure you\'re using the AMD 64 image (ubuntu-13.04-desktop-amd64.iso) and not the x86 image. That way, you get EFI boot capability.\n\n&nbsp;\n\nWill expand on this further at a later time.','Installing Ubuntu 13.04 on Dell XPS 14','','inherit','open','open','','371-autosave','','','2013-10-21 21:07:21','2013-10-22 01:07:21','',371,'http://www.andrewychoi.com/2013/371-autosave/',0,'revision','',0),
(401,1,'2013-10-22 02:14:28','2013-10-22 06:14:28','I ran the following commands to get things running:\r\n\r\n&nbsp;\r\n\r\nsudo aptitude install hal\r\n\r\nsudo mkdir /etc/hal/fdi/preprobe\r\nsudo mkdir /etc/hal/fdi/information\r\n/usr/sbin/hald --daemon=yes --verbose=yes\r\nrm -rf ~/.adobe','Getting Amazon Instant Video working on Ubuntu 13.04 and Firefox','','publish','open','open','','getting-amazon-instant-video-working-on-ubuntu-13-04-and-firefox','','','2013-10-22 02:14:28','2013-10-22 06:14:28','',0,'http://www.andrewychoi.com/?p=401',0,'post','',0),
(411,1,'2013-10-22 02:14:00','2013-10-22 06:14:00','','Getting Amazon Instant Video working on Ubuntu 13.04 and Firefox','','inherit','open','open','','401-revision','','','2013-10-22 02:14:00','2013-10-22 06:14:00','',401,'http://www.andrewychoi.com/2013/401-revision/',0,'revision','',0),
(421,1,'2014-06-12 17:34:15','2014-06-12 21:34:15','If you\'re using <code>mchung</code>\'s Heroku buildpack for WordPress (located at <a title=\"https://github.com/mchung/heroku-buildpack-wordpress\" href=\"https://github.com/mchung/heroku-buildpack-wordpress\">https://github.com/mchung/heroku-buildpack-wordpress</a>), like I am, it may be non-obvious how to update your WordPress install. There are two main steps that you need to take.\r\n\r\nFirst, in the Heroku dashboard, under Settings, click on \"Show config vars\", and then \"Edit\". Change <code>WORDPRESS_VERSION</code> to the latest (as of right now, 3.9.1), and then click save.\r\n\r\nThen, to trigger a rebuild, push an empty commit to your repository like so:\r\n<pre>git commit --allow-empty -m \"empty commit\"\r\ngit push heroku master</pre>\r\nThen, when prompted, update your database, and you\'re done!','Updating the Wordpress version on Heroku','','publish','open','open','','updating-the-wordpress-version-on-heroku','','','2014-06-12 17:34:16','2014-06-12 21:34:16','',0,'http://www.andrewychoi.com/?p=421',0,'post','',0),
(431,1,'2014-06-12 17:34:16','2014-06-12 21:34:16','If you\'re using <code>mchung</code>\'s Heroku buildpack for WordPress (located at <a title=\"https://github.com/mchung/heroku-buildpack-wordpress\" href=\"https://github.com/mchung/heroku-buildpack-wordpress\">https://github.com/mchung/heroku-buildpack-wordpress</a>), like I am, it may be non-obvious how to update your WordPress install. There are two main steps that you need to take.\r\n\r\nFirst, in the Heroku dashboard, under Settings, click on \"Show config vars\", and then \"Edit\". Change <code>WORDPRESS_VERSION</code> to the latest (as of right now, 3.9.1), and then click save.\r\n\r\nThen, to trigger a rebuild, push an empty commit to your repository like so:\r\n<pre>git commit --allow-empty -m \"empty commit\"\r\ngit push heroku master</pre>\r\nThen, when prompted, update your database, and you\'re done!','Updating the Wordpress version on Heroku','','inherit','open','open','','421-revision-v1','','','2014-06-12 17:34:16','2014-06-12 21:34:16','',421,'http://www.andrewychoi.com/2014/421-revision-v1/',0,'revision','',0),
(441,1,'2014-06-12 17:47:28','2014-06-12 21:47:28','I\'m currently on a Verizon FiOS connection here at home, and I found that when entering one word terms into my URL bar in Firefox, I would get redirected to <code>search.dnsassist.verizon.com</code> for search results. Interestingly, this did not happen when I entered terms separated by spaces. My intuition led me to believe that this was a DNS resolution service, that was somehow not getting triggered by things that Firefox didn\'t believe were valid URLs. To fix these results, I changed my IPv4 settings on my (now 14.04 LTS, thank goodness) Ubuntu install to use DHCP only for local addressing. I changed my DNS records to point at Google\'s public facing DNS servers, at <code>8.8.8.8</code> and <code>8.8.4.4</code>. Now, since my computer is no longer looking on the local network for a DNS server (and thus not getting results from Verizon\'s servers) I can enter a single term and get my Google search!\r\n\r\nA note: those public-facing DNS server IP addresses are good to memorize, in case you get into a situation where you\'re missing a DNS server. This actually happened to me one time in 2010, when I was working as an IT technician at a hotel conference. The IPv6 addresses are a bit harder to memorize, but they can all be found <a href=\"https://developers.google.com/speed/public-dns/docs/using\">here on Google\'s site</a>.','Removing Verizon DNS Search Assist','','publish','open','open','','removing-verizon-dns-search-assist','','','2014-06-12 17:47:28','2014-06-12 21:47:28','',0,'http://www.andrewychoi.com/?p=441',0,'post','',3),
(451,1,'2014-06-12 17:47:28','2014-06-12 21:47:28','I\'m currently on a Verizon FiOS connection here at home, and I found that when entering one word terms into my URL bar in Firefox, I would get redirected to <code>search.dnsassist.verizon.com</code> for search results. Interestingly, this did not happen when I entered terms separated by spaces. My intuition led me to believe that this was a DNS resolution service, that was somehow not getting triggered by things that Firefox didn\'t believe were valid URLs. To fix these results, I changed my IPv4 settings on my (now 14.04 LTS, thank goodness) Ubuntu install to use DHCP only for local addressing. I changed my DNS records to point at Google\'s public facing DNS servers, at <code>8.8.8.8</code> and <code>8.8.4.4</code>. Now, since my computer is no longer looking on the local network for a DNS server (and thus not getting results from Verizon\'s servers) I can enter a single term and get my Google search!\r\n\r\nA note: those public-facing DNS server IP addresses are good to memorize, in case you get into a situation where you\'re missing a DNS server. This actually happened to me one time in 2010, when I was working as an IT technician at a hotel conference. The IPv6 addresses are a bit harder to memorize, but they can all be found <a href=\"https://developers.google.com/speed/public-dns/docs/using\">here on Google\'s site</a>.','Removing Verizon DNS Search Assist','','inherit','open','open','','441-revision-v1','','','2014-06-12 17:47:28','2014-06-12 21:47:28','',441,'http://www.andrewychoi.com/2014/441-revision-v1/',0,'revision','',0),
(461,1,'2014-06-12 18:02:06','2014-06-12 22:02:06','Current Projects:\n\nPython 3 implementation of LinkedIn API:\n\nPast Projects:\n\nPrinceton International Relations - September 2010 - June 2013\n\nPrinceton ELE 381 - 20 Questions About Our Networked Life - April 2011 - June 2013\n\nPrinceton EDGE Lab - Web Developer - June 2012 - June 2013\n\nQuest Scholars Network - June - August 2011','Projects','','inherit','open','open','','5-autosave-v1','','','2014-06-12 18:02:06','2014-06-12 22:02:06','',5,'http://www.andrewychoi.com/2014/5-autosave-v1/',0,'revision','',0),
(471,1,'2014-06-12 18:02:38','2014-06-12 22:02:38','Current Projects:\r\n\r\n<a href=\"https://github.com/andrewychoi/python3-linkedin\">Python 3 implementation of LinkedIn API</a>\r\n\r\nPast Projects:\r\n\r\nPrinceton International Relations - September 2010 - June 2013\r\n\r\nPrinceton ELE 381 - 20 Questions About Our Networked Life - April 2011 - June 2013\r\n\r\nPrinceton EDGE Lab - Web Developer - June 2012 - June 2013\r\n\r\nQuest Scholars Network - June - August 2011','Projects','','inherit','open','open','','5-revision-v1','','','2014-06-12 18:02:38','2014-06-12 22:02:38','',5,'http://www.andrewychoi.com/2014/5-revision-v1/',0,'revision','',0),
(481,1,'2014-07-10 12:15:02','2014-07-10 16:15:02','Today I decided I needed to print something, and for the first time I had to configure a printer to use in Ubuntu. I connected my Brother HL 2140 via USB cable and tried to print, only to find that the printer hadn\'t been added (I\'ve been spoiled by Windows, I guess). I then went to System Settings -&gt; Printers -&gt; Add, and connected the Brother printer that was attached via USB. Tried printing then, and blank page after blank page kept coming out! I cancelled the print job, and realized that the printer couldn\'t handle PostScript. After messing around with a few other drivers, I found that the \"Brother HL-2140 Foomatic/hl1250\" driver worked for me. If I have to print something else out, I may try experimenting with others, but since this works for me right now, I won\'t try to fix it any further.','Printing blank pages with Brother HL 2140 and Ubuntu 14.04','','publish','open','open','','printing-blank-pages-with-brother-hl-2140-and-ubuntu-14-04','','','2014-07-10 12:15:02','2014-07-10 16:15:02','',0,'http://www.andrewychoi.com/?p=481',0,'post','',3),
(491,1,'2014-07-10 12:15:02','2014-07-10 16:15:02','Today I decided I needed to print something, and for the first time I had to configure a printer to use in Ubuntu. I connected my Brother HL 2140 via USB cable and tried to print, only to find that the printer hadn\'t been added (I\'ve been spoiled by Windows, I guess). I then went to System Settings -&gt; Printers -&gt; Add, and connected the Brother printer that was attached via USB. Tried printing then, and blank page after blank page kept coming out! I cancelled the print job, and realized that the printer couldn\'t handle PostScript. After messing around with a few other drivers, I found that the \"Brother HL-2140 Foomatic/hl1250\" driver worked for me. If I have to print something else out, I may try experimenting with others, but since this works for me right now, I won\'t try to fix it any further.','Printing blank pages with Brother HL 2140 and Ubuntu 14.04','','inherit','open','open','','481-revision-v1','','','2014-07-10 12:15:02','2014-07-10 16:15:02','',481,'http://www.andrewychoi.com/2014/481-revision-v1/',0,'revision','',0),
(501,1,'2014-07-12 14:51:04','2014-07-12 18:51:04','I ran into this error today during development:\r\n<code>\r\nImproperlyConfigured at /project/path/here/\r\nYou\'re using the staticfiles app without having set the\r\nSTATIC_ROOT setting to a filesystem path.\r\n</code>\r\nI\'m running Django 1.6 on Ubuntu 14.04. If you\'re getting this error when you include something like the following in your template code:\r\n\r\n<code>{% load compressed %}\r\n...\r\n{% compressed_css \"bootstrap\" %}</code>\r\n\r\nMake sure you\'re using parentheses, not curly braces in your glob syntax for the <code>source_filenames</code> setting:\r\n\r\nThus,\r\n<pre>\"bootstrap\": {\r\n    \"source_filenames\": {\r\n        \"libs/bootstrap/css/boostrap.min.css\",\r\n    },\r\n    \"output_filename\": \"pipeline/css/bootstrap.css\",\r\n},\r\n</pre>\r\nbecomes\r\n<pre>\"bootstrap\": {\r\n    \"source_filenames\": (\r\n        \"libs/bootstrap/css/boostrap.min.css\",\r\n    ),\r\n    \"output_filename\": \"pipeline/css/bootstrap.css\",\r\n},\r\n</pre>\r\n\r\nSmall error, and it\'s tough to find. The error message isn\'t particularly helpful, either.','ImproperlyConfigured error using django-pipeline ','','publish','open','open','','improperlyconfigured-error-using-django-pipeline','','','2014-07-12 14:52:30','2014-07-12 18:52:30','',0,'http://www.andrewychoi.com/?p=501',0,'post','',0),
(511,1,'2014-07-12 14:51:04','2014-07-12 18:51:04','I ran into this error today during development:\r\n<code>\r\nImproperlyConfigured at /project/path/here/\r\nYou\'re using the staticfiles app without having set the\r\nSTATIC_ROOT setting to a filesystem path.\r\n</code>\r\nI\'m running Django 1.6 on Ubuntu 14.04. If you\'re getting this error when you include something like the following in your template code:\r\n\r\n<code>{% load compressed %}\r\n...\r\n{% compressed_css \"bootstrap\" %}</code>\r\n\r\nMake sure you\'re using parentheses, not curly braces in your glob syntax for the <code>source_filenames</code> setting:\r\n\r\nThus,\r\n<code>    \"bootstrap\": {\r\n        \"source_filenames\": {\r\n            \"libs/bootstrap/css/boostrap.min.css\",\r\n        },\r\n        \"output_filename\": \"pipeline/css/bootstrap.css\",\r\n    },</code>\r\nbecomes\r\n<code>    \"bootstrap\": {\r\n        \"source_filenames\": (\r\n            \"libs/bootstrap/css/boostrap.min.css\",\r\n        ),\r\n        \"output_filename\": \"pipeline/css/bootstrap.css\",\r\n    },\r\n</code>','ImproperlyConfigured error using django-pipeline ','','inherit','open','open','','501-revision-v1','','','2014-07-12 14:51:04','2014-07-12 18:51:04','',501,'http://www.andrewychoi.com/2014/501-revision-v1/',0,'revision','',0),
(521,1,'2014-07-12 14:52:10','2014-07-12 18:52:10','I ran into this error today during development:\n<code>\nImproperlyConfigured at /project/path/here/\nYou\'re using the staticfiles app without having set the\nSTATIC_ROOT setting to a filesystem path.\n</code>\nI\'m running Django 1.6 on Ubuntu 14.04. If you\'re getting this error when you include something like the following in your template code:\n\n<code>{% load compressed %}\n...\n{% compressed_css \"bootstrap\" %}</code>\n\nMake sure you\'re using parentheses, not curly braces in your glob syntax for the <code>source_filenames</code> setting:\n\nThus,\n<pre>\"bootstrap\": {\n    \"source_filenames\": {\n        \"libs/bootstrap/css/boostrap.min.css\",\n    },\n    \"output_filename\": \"pipeline/css/bootstrap.css\",\n},\n</pre>\nbecomes\n<pre>\"bootstrap\": {\n    \"source_filenames\": (\n        \"libs/bootstrap/css/boostrap.min.css\",\n    ),\n    \"output_filename\": \"pipeline/css/bootstrap.css\",\n},\n</pre>\n\nSmall error, and it\'','ImproperlyConfigured error using django-pipeline ','','inherit','open','open','','501-autosave-v1','','','2014-07-12 14:52:10','2014-07-12 18:52:10','',501,'http://www.andrewychoi.com/2014/501-autosave-v1/',0,'revision','',0),
(531,1,'2014-07-12 14:52:30','2014-07-12 18:52:30','I ran into this error today during development:\r\n<code>\r\nImproperlyConfigured at /project/path/here/\r\nYou\'re using the staticfiles app without having set the\r\nSTATIC_ROOT setting to a filesystem path.\r\n</code>\r\nI\'m running Django 1.6 on Ubuntu 14.04. If you\'re getting this error when you include something like the following in your template code:\r\n\r\n<code>{% load compressed %}\r\n...\r\n{% compressed_css \"bootstrap\" %}</code>\r\n\r\nMake sure you\'re using parentheses, not curly braces in your glob syntax for the <code>source_filenames</code> setting:\r\n\r\nThus,\r\n<pre>\"bootstrap\": {\r\n    \"source_filenames\": {\r\n        \"libs/bootstrap/css/boostrap.min.css\",\r\n    },\r\n    \"output_filename\": \"pipeline/css/bootstrap.css\",\r\n},\r\n</pre>\r\nbecomes\r\n<pre>\"bootstrap\": {\r\n    \"source_filenames\": (\r\n        \"libs/bootstrap/css/boostrap.min.css\",\r\n    ),\r\n    \"output_filename\": \"pipeline/css/bootstrap.css\",\r\n},\r\n</pre>\r\n\r\nSmall error, and it\'s tough to find. The error message isn\'t particularly helpful, either.','ImproperlyConfigured error using django-pipeline ','','inherit','open','open','','501-revision-v1','','','2014-07-12 14:52:30','2014-07-12 18:52:30','',501,'http://www.andrewychoi.com/2014/501-revision-v1/',0,'revision','',0),
(541,1,'2014-07-31 15:03:14','2014-07-31 19:03:14','Today I deleted a branch on a remote repo via the web interface, and was faced with the issue of my local repo not being updated for the changes. <code>git pull</code> and <code>git push</code> did nothing, unfortunately, and most of the solutions that I found online involved a <code>git push origin :branch_name</code> or its variants. When running this command, however, I got the following error message:\r\n<code>\r\n$ git push origin :branch_name\r\nerror: unable to delete \'branch_name\': remote ref does not exist\r\nerror: failed to push some refs to \'git@bitbucket.org:andrewchoi/repo_name.git\'\r\n</code>\r\nThankfully, after a good amount of Google-fu, I found the <code>git remote prune</code> command, which I ran on the remote <code>origin</code> like so:\r\n<code>\r\n$ git remote prune origin\r\n</code>\r\nwhich solved my issue.','git remote prune','','publish','open','open','','git-remote-prune','','','2014-07-31 15:03:14','2014-07-31 19:03:14','',0,'http://www.andrewychoi.com/?p=541',0,'post','',0),
(551,1,'2014-07-31 15:03:14','2014-07-31 19:03:14','Today I deleted a branch on a remote repo via the web interface, and was faced with the issue of my local repo not being updated for the changes. <code>git pull</code> and <code>git push</code> did nothing, unfortunately, and most of the solutions that I found online involved a <code>git push origin :branch_name</code> or its variants. When running this command, however, I got the following error message:\r\n<code>\r\n$ git push origin :branch_name\r\nerror: unable to delete \'branch_name\': remote ref does not exist\r\nerror: failed to push some refs to \'git@bitbucket.org:andrewchoi/repo_name.git\'\r\n</code>\r\nThankfully, after a good amount of Google-fu, I found the <code>git remote prune</code> command, which I ran on the remote <code>origin</code> like so:\r\n<code>\r\n$ git remote prune origin\r\n</code>\r\nwhich solved my issue.','git remote prune','','inherit','open','open','','541-revision-v1','','','2014-07-31 15:03:14','2014-07-31 19:03:14','',541,'http://www.andrewychoi.com/2014/541-revision-v1/',0,'revision','',0),
(561,1,'2014-08-12 11:52:33','2014-08-12 15:52:33','I was having an issue with Distraction Free mode with showing line numbers. I went to the user-specified Distraction Free settings (Preferences -&gt; Settings - More -&gt; Distraction Free - User) and tried to change the setting<code>\"line_numbers\": true</code>. However, when I went into Distraction Free, there were no line numbers. I tried restarting ST3 and even my whole computer, but to no avail. After some Googling, I found that to display line numbers, you have to change not just the <code>line_numbers</code> setting but also the <code>gutter</code> setting to be <code>true</code>. My user-specific Distraction Free settings now looks like:\r\n<pre>{\r\n    \"line_numbers\": true,\r\n    \"gutter\": true,\r\n    // \"line_numbers\": false,\r\n    // \"gutter\": false,\r\n    \"draw_centered\": true,\r\n    \"wrap_width\": 82,\r\n    \"word_wrap\": true,\r\n    \"scroll_past_end\": true,\r\n    \"rulers\": [],\r\n}\r\n</pre>','Line numbers not showing in Distraction Free on Sublime Text 3','','publish','open','open','','line-numbers-not-showing-in-distraction-free-on-sublime-text-3','','','2014-08-12 11:52:33','2014-08-12 15:52:33','',0,'http://www.andrewychoi.com/?p=561',0,'post','',1),
(571,1,'2014-08-12 11:52:33','2014-08-12 15:52:33','I was having an issue with Distraction Free mode with showing line numbers. I went to the user-specified Distraction Free settings (Preferences -&gt; Settings - More -&gt; Distraction Free - User) and tried to change the setting<code>\"line_numbers\": true</code>. However, when I went into Distraction Free, there were no line numbers. I tried restarting ST3 and even my whole computer, but to no avail. After some Googling, I found that to display line numbers, you have to change not just the <code>line_numbers</code> setting but also the <code>gutter</code> setting to be <code>true</code>. My user-specific Distraction Free settings now looks like:\r\n<pre>{\r\n    \"line_numbers\": true,\r\n    \"gutter\": true,\r\n    // \"line_numbers\": false,\r\n    // \"gutter\": false,\r\n    \"draw_centered\": true,\r\n    \"wrap_width\": 82,\r\n    \"word_wrap\": true,\r\n    \"scroll_past_end\": true,\r\n    \"rulers\": [],\r\n}\r\n</pre>','Line numbers not showing in Distraction Free on Sublime Text 3','','inherit','open','open','','561-revision-v1','','','2014-08-12 11:52:33','2014-08-12 15:52:33','',561,'http://www.andrewychoi.com/2014/561-revision-v1/',0,'revision','',0),
(581,1,'2014-08-14 21:25:39','2014-08-15 01:25:39','A quick gotcha that I found today:\r\n\r\nWhile you can compare <code>int</code> values with other <code>int</code> values and it works as expected, the same is not true of many other types. I was aware of the fact that you should use <code>!=</code> for string comparison (because when you use <code>is</code>, Python checks to see if the address of the underlying string is the same), but I just found out that you have to use <code>!=</code> for comparison of floats. It seems that unlike <code>int</code> values, where Python has a single object per value, <code>float</code> values are different!','a is not b vs a != b for float comparison in Python','','publish','open','open','','a-is-not-b-vs-a-b-for-float-comparison-in-python','','','2014-08-14 21:25:39','2014-08-15 01:25:39','',0,'http://www.andrewychoi.com/?p=581',0,'post','',0),
(591,1,'2014-08-14 21:25:39','2014-08-15 01:25:39','A quick gotcha that I found today:\r\n\r\nWhile you can compare <code>int</code> values with other <code>int</code> values and it works as expected, the same is not true of many other types. I was aware of the fact that you should use <code>!=</code> for string comparison (because when you use <code>is</code>, Python checks to see if the address of the underlying string is the same), but I just found out that you have to use <code>!=</code> for comparison of floats. It seems that unlike <code>int</code> values, where Python has a single object per value, <code>float</code> values are different!','a is not b vs a != b for float comparison in Python','','inherit','open','open','','581-revision-v1','','','2014-08-14 21:25:39','2014-08-15 01:25:39','',581,'http://www.andrewychoi.com/2014/581-revision-v1/',0,'revision','',0),
(601,1,'2014-08-19 15:27:40','2014-08-19 19:27:40','Today I needed to set up a local copy of the Postgres database that we have on a Heroku instance somewhere in the cloud for testing and debugging purposes on my local machine. Steps taken:\r\n<ul>\r\n	<li>Create a new Postgres user:\r\n<pre>&gt; psql -d postgres # default user that postgres creates when you install it\r\npostgres=# create user your_username;\r\n</pre>\r\n</li>\r\n	<li>Create a database for your app:\r\n<pre>postgres=# create database your_db_name;\r\n</pre>\r\n</li>\r\n	<li>Grant the user permissions to create databases (this is necessary for running Django tests):\r\n<pre>postgres=# alter user your_username createdb;\r\n</pre>\r\n</li>\r\n<li>\r\nDump the data from your Heroku backup (you did download that already, right?)\r\n<pre>\r\npg_restore --verbose --clean --no-acl --no-owner -h localhost -U your_username -d your_dbname /path/to/your/backup.dump \r\n</pre>\r\n</ul>\r\n\r\nAnd now you should have a local copy of your DB, as well as the ability to run tests!','Setting up a local Postgres environment','','publish','open','open','','setting-up-a-local-postgres-environment','','','2014-08-19 15:27:40','2014-08-19 19:27:40','',0,'http://www.andrewychoi.com/?p=601',0,'post','',0),
(611,1,'2014-08-19 15:27:40','2014-08-19 19:27:40','Today I needed to set up a local copy of the Postgres database that we have on a Heroku instance somewhere in the cloud for testing and debugging purposes on my local machine. Steps taken:\r\n<ul>\r\n	<li>Create a new Postgres user:\r\n<pre>&gt; psql -d postgres # default user that postgres creates when you install it\r\npostgres=# create user your_username;\r\n</pre>\r\n</li>\r\n	<li>Create a database for your app:\r\n<pre>postgres=# create database your_db_name;\r\n</pre>\r\n</li>\r\n	<li>Grant the user permissions to create databases (this is necessary for running Django tests):\r\n<pre>postgres=# alter user your_username createdb;\r\n</pre>\r\n</li>\r\n<li>\r\nDump the data from your Heroku backup (you did download that already, right?)\r\n<pre>\r\npg_restore --verbose --clean --no-acl --no-owner -h localhost -U your_username -d your_dbname /path/to/your/backup.dump \r\n</pre>\r\n</ul>\r\n\r\nAnd now you should have a local copy of your DB, as well as the ability to run tests!','Setting up a local Postgres environment','','inherit','open','open','','601-revision-v1','','','2014-08-19 15:27:40','2014-08-19 19:27:40','',601,'http://www.andrewychoi.com/2014/601-revision-v1/',0,'revision','',0),
(621,1,'2014-08-25 12:37:48','2014-08-25 16:37:48','I\'m building a tool to export results from a Django ORM query to JSON, and today I ran into a bug that would have taken a whole lot longer to debug had I not known about iterators and how they work. (For a primer, check out https://docs.python.org/3/tutorial/classes.html#iterators or something similar). My code had a section in it where I use the groupby() function in the itertools package to group results so that I can represent them hierarchically. So, my code looked a bit like this:\r\n<pre>\r\nsave_dict = {}\r\nfor key, group in itertools.groupby(thing_to_group, key = lambda x: x[\"group_key\"]):\r\nprint key\r\nfor item in group:\r\nprint item\r\nsave_dict[key] = {item[\"subkey\"]: item[\"value\"] for item in group}\r\n</pre>\r\nBut when I printed out my results, I found:\r\n<pre>\r\n> save_dict[key]\r\n{}\r\n</pre>\r\nThe dictionary was empty!\r\n\r\nThe issue with my code was that by going through the group iterator already, I had \"exhausted\" the iterator, and so the dict comprehension was operating on an empty iterable. Removing the offending value checking code solved the problem!\r\n\r\nFor future reference, if you need to retain the values from the iterable, use the handy list() function (taken from: https://docs.python.org/2/library/itertools.html#itertools.groupby) like so:\r\n<pre>\r\nfor key, group in itertools.groupby(thing_to_group, key = lambda x: x[\"group_key\"]):\r\nsaved_group = list(group)\r\nfor item in saved_group:\r\nprint item # you get to eat all your values\r\nfor item in saved_group:\r\nprint item # and have them too!\r\nsave_dict[key] = {item[\"subkey\"]: item[\"value\"] for item in group} # this will work now\r\n</pre>','Iterators: not even twice','','publish','open','open','','iterators-not-even-twice','','','2014-08-25 12:37:48','2014-08-25 16:37:48','',0,'http://www.andrewychoi.com/?p=621',0,'post','',0),
(631,1,'2014-08-25 12:37:48','2014-08-25 16:37:48','I\'m building a tool to export results from a Django ORM query to JSON, and today I ran into a bug that would have taken a whole lot longer to debug had I not known about iterators and how they work. (For a primer, check out https://docs.python.org/3/tutorial/classes.html#iterators or something similar). My code had a section in it where I use the groupby() function in the itertools package to group results so that I can represent them hierarchically. So, my code looked a bit like this:\r\n<pre>\r\nsave_dict = {}\r\nfor key, group in itertools.groupby(thing_to_group, key = lambda x: x[\"group_key\"]):\r\nprint key\r\nfor item in group:\r\nprint item\r\nsave_dict[key] = {item[\"subkey\"]: item[\"value\"] for item in group}\r\n</pre>\r\nBut when I printed out my results, I found:\r\n<pre>\r\n> save_dict[key]\r\n{}\r\n</pre>\r\nThe dictionary was empty!\r\n\r\nThe issue with my code was that by going through the group iterator already, I had \"exhausted\" the iterator, and so the dict comprehension was operating on an empty iterable. Removing the offending value checking code solved the problem!\r\n\r\nFor future reference, if you need to retain the values from the iterable, use the handy list() function (taken from: https://docs.python.org/2/library/itertools.html#itertools.groupby) like so:\r\n<pre>\r\nfor key, group in itertools.groupby(thing_to_group, key = lambda x: x[\"group_key\"]):\r\nsaved_group = list(group)\r\nfor item in saved_group:\r\nprint item # you get to eat all your values\r\nfor item in saved_group:\r\nprint item # and have them too!\r\nsave_dict[key] = {item[\"subkey\"]: item[\"value\"] for item in group} # this will work now\r\n</pre>','Iterators: not even twice','','inherit','open','open','','621-revision-v1','','','2014-08-25 12:37:48','2014-08-25 16:37:48','',621,'http://www.andrewychoi.com/2014/621-revision-v1/',0,'revision','',0),
(641,1,'2014-09-04 10:07:07','2014-09-04 14:07:07','If you\'re getting a failure along these lines:\r\n\r\n<code>\r\ncompiz crashed with SIGSEGV in ScaleScreen::getSelectedWindow()\r\n</code>\r\n\r\nit seems to be triggered by having multiple windows of the same application open, and then clicking on them in the launcher. This runs the screen spread feature, which is what is causing the crash. I found a fix here: http://speculator.diandian.com/post/2014-07-01/40062129998 but the skinny is that I ran\r\n\r\n<code>\r\ndconf reset -f /org/compiz/\r\n</code>\r\nand rebooted. According to that blog post, the instructions were found here: http://www.omgubuntu.co.uk/2013/04/how-to-reset-unity-compiz-in-ubuntu-12-10-and-13-04.','Compiz crashing when clicking icon in launcher: Ubuntu 14.04','','publish','open','open','','compiz-crashing-when-clicking-icon-in-launcher-ubuntu-14-04','','','2014-09-04 10:07:07','2014-09-04 14:07:07','',0,'http://www.andrewychoi.com/?p=641',0,'post','',1),
(651,1,'2014-09-04 10:07:07','2014-09-04 14:07:07','If you\'re getting a failure along these lines:\r\n\r\n<code>\r\ncompiz crashed with SIGSEGV in ScaleScreen::getSelectedWindow()\r\n</code>\r\n\r\nit seems to be triggered by having multiple windows of the same application open, and then clicking on them in the launcher. This runs the screen spread feature, which is what is causing the crash. I found a fix here: http://speculator.diandian.com/post/2014-07-01/40062129998 but the skinny is that I ran\r\n\r\n<code>\r\ndconf reset -f /org/compiz/\r\n</code>\r\nand rebooted. According to that blog post, the instructions were found here: http://www.omgubuntu.co.uk/2013/04/how-to-reset-unity-compiz-in-ubuntu-12-10-and-13-04.','Compiz crashing when clicking icon in launcher: Ubuntu 14.04','','inherit','open','open','','641-revision-v1','','','2014-09-04 10:07:07','2014-09-04 14:07:07','',641,'http://www.andrewychoi.com/2014/641-revision-v1/',0,'revision','',0),
(661,1,'2014-09-22 15:28:28','0000-00-00 00:00:00','','Auto Draft','','auto-draft','open','open','','','','','2014-09-22 15:28:28','0000-00-00 00:00:00','',0,'http://www.andrewychoi.com/?p=661',0,'post','',0),
(671,1,'2014-09-22 15:30:57','2014-09-22 19:30:57','If you\'re getting something like:\r\n\r\n<code>IntegrityError: duplicate key value violates unique constraint \"table_pkey\"\r\nDETAIL:  Key (id)=(3455) already exists.</code>\r\n\r\nand you\'ve just imported a database into an existing install, your database may not have the correct autoincrement value in place. In postgres, the following will fix the issue:\r\n\r\n<code>SELECT setval(\'yourtable_id_seq\', (SELECT MAX(id) from yourtable));</code>\r\n\r\nFrom: http://stackoverflow.com/questions/9108833/postgres-autoincrement-not-updated-on-explicit-id-inserts\r\n','IntegrityError after importing database','','publish','open','open','','integrityerror-after-importing-database','','','2014-09-22 15:30:57','2014-09-22 19:30:57','',0,'http://www.andrewychoi.com/?p=671',0,'post','',0),
(681,1,'2014-09-22 15:30:57','2014-09-22 19:30:57','If you\'re getting something like:\r\n\r\n<code>IntegrityError: duplicate key value violates unique constraint \"table_pkey\"\r\nDETAIL:  Key (id)=(3455) already exists.</code>\r\n\r\nand you\'ve just imported a database into an existing install, your database may not have the correct autoincrement value in place. In postgres, the following will fix the issue:\r\n\r\n<code>SELECT setval(\'yourtable_id_seq\', (SELECT MAX(id) from yourtable));</code>\r\n\r\nFrom: http://stackoverflow.com/questions/9108833/postgres-autoincrement-not-updated-on-explicit-id-inserts\r\n','IntegrityError after importing database','','inherit','open','open','','671-revision-v1','','','2014-09-22 15:30:57','2014-09-22 19:30:57','',671,'http://www.andrewychoi.com/2014/671-revision-v1/',0,'revision','',0),
(691,1,'2014-09-29 14:11:19','2014-09-29 18:11:19','I was having an issue using Django admin form widgets outside of the actual admin site, and found out that my call to form.media[\"js\"] needed to be in the <head> and not at the end of <body> as is normally advised. Otherwise, you get a\r\naddEvent is undefined error or something similar.','Using Django admin form widgets','','publish','open','open','','using-django-admin-form-widgets','','','2014-09-29 14:11:19','2014-09-29 18:11:19','',0,'http://www.andrewychoi.com/?p=691',0,'post','',0),
(701,1,'2014-09-29 14:11:19','2014-09-29 18:11:19','I was having an issue using Django admin form widgets outside of the actual admin site, and found out that my call to form.media[\"js\"] needed to be in the <head> and not at the end of <body> as is normally advised. Otherwise, you get a\r\naddEvent is undefined error or something similar.','Using Django admin form widgets','','inherit','open','open','','691-revision-v1','','','2014-09-29 14:11:19','2014-09-29 18:11:19','',691,'http://www.andrewychoi.com/2014/691-revision-v1/',0,'revision','',0),
(711,1,'2014-10-10 13:31:24','2014-10-10 17:31:24','Today I was working on toggling classes of elements in a d3-generated SVG, and addClass and removeClass weren\'t working. I verified that my selectors were correct, and also surrounded the block with a jQuery document-ready function, but no dice. I switched from using the jQuery addClass and removeClass methods to d3\'s classed() function, and the code magically started working!\r\n\r\nAs a further note, I imagine that the reason why this is the case is that addClass and removeClass don\'t recognize <path> elements, while the selectors still do.','Classes with d3 SVGs','','publish','open','open','','classes-with-d3-svgs','','','2014-10-10 13:31:24','2014-10-10 17:31:24','',0,'http://www.andrewychoi.com/?p=711',0,'post','',0),
(721,1,'2014-10-10 13:31:24','2014-10-10 17:31:24','Today I was working on toggling classes of elements in a d3-generated SVG, and addClass and removeClass weren\'t working. I verified that my selectors were correct, and also surrounded the block with a jQuery document-ready function, but no dice. I switched from using the jQuery addClass and removeClass methods to d3\'s classed() function, and the code magically started working!\r\n\r\nAs a further note, I imagine that the reason why this is the case is that addClass and removeClass don\'t recognize <path> elements, while the selectors still do.','Classes with d3 SVGs','','inherit','open','open','','711-revision-v1','','','2014-10-10 13:31:24','2014-10-10 17:31:24','',711,'http://www.andrewychoi.com/2014/711-revision-v1/',0,'revision','',0),
(731,1,'2014-11-05 12:05:34','2014-11-05 17:05:34','andrew (at) choi (dot) com','Contact','','inherit','open','open','','51-revision-v1','','','2014-11-05 12:05:34','2014-11-05 17:05:34','',51,'http://www.andrewychoi.com/2014/51-revision-v1/',0,'revision','',0),
(741,1,'2014-12-04 11:23:45','2014-12-04 16:23:45','Be aware: if you\'re trying to use a custom model manager in Django, any inclusion of a manager will remove the default <pre>objects</pre> manager. Be sure to re-include it if you\'re just extending your models!','Custom model Managers in Django','','publish','open','open','','custom-model-managers-in-django','','','2014-12-04 11:23:45','2014-12-04 16:23:45','',0,'http://www.andrewychoi.com/?p=741',0,'post','',0),
(751,1,'2014-12-04 11:23:45','2014-12-04 16:23:45','Be aware: if you\'re trying to use a custom model manager in Django, any inclusion of a manager will remove the default <pre>objects</pre> manager. Be sure to re-include it if you\'re just extending your models!','Custom model Managers in Django','','inherit','open','open','','741-revision-v1','','','2014-12-04 11:23:45','2014-12-04 16:23:45','',741,'http://www.andrewychoi.com/2014/741-revision-v1/',0,'revision','',0),
(761,1,'2015-03-23 13:44:56','2015-03-23 17:44:56','<pre>ImportError: ...lib/python2.7/site-packages/numpy/core/multiarray.so: undefined symbol: PyUnicodeUCS4_AsUnicodeEscapeString</pre>\r\nThe above error means that you\'ve got a Python interpreter compiled against UCS2 trying to run code that was compiled against UCS4. The solution I found was to recompile your lib (here NumPy) using a Python binary compiled with a\r\n<pre>--enable-unicode=ucs2</pre>\r\n.','ImportError: undefined symbol: PyUnicodeUCS4_AsUnicodeEscapeString','','publish','open','open','','importerror-undefined-symbol-pyunicodeucs4_asunicodeescapestring','','','2015-03-23 13:45:16','2015-03-23 17:45:16','',0,'http://www.andrewychoi.com/?p=761',0,'post','',0),
(771,1,'2015-03-23 13:44:56','2015-03-23 17:44:56','<pre>ImportError: ...lib/python2.7/site-packages/numpy/core/multiarray.so: undefined symbol: PyUnicodeUCS4_AsUnicodeEscapeString</pre>\r\nThe above error means that you\'ve got a Python interpreter compiled against UCS2 trying to run code that was compiled against UCS4. The solution I found was to recompile your lib (here NumPy) using a Python binary compiled with a\r\n<pre>--enable-unicode=UCS2</pre>\r\n.','ImportError: undefined symbol: PyUnicodeUCS4_AsUnicodeEscapeString','','inherit','open','open','','761-revision-v1','','','2015-03-23 13:44:56','2015-03-23 17:44:56','',761,'http://www.andrewychoi.com/2015/761-revision-v1/',0,'revision','',0),
(781,1,'2015-03-23 13:45:16','2015-03-23 17:45:16','<pre>ImportError: ...lib/python2.7/site-packages/numpy/core/multiarray.so: undefined symbol: PyUnicodeUCS4_AsUnicodeEscapeString</pre>\r\nThe above error means that you\'ve got a Python interpreter compiled against UCS2 trying to run code that was compiled against UCS4. The solution I found was to recompile your lib (here NumPy) using a Python binary compiled with a\r\n<pre>--enable-unicode=ucs2</pre>\r\n.','ImportError: undefined symbol: PyUnicodeUCS4_AsUnicodeEscapeString','','inherit','open','open','','761-revision-v1','','','2015-03-23 13:45:16','2015-03-23 17:45:16','',761,'http://www.andrewychoi.com/2015/761-revision-v1/',0,'revision','',0);
